src/fastfield/multivalued/writer.rs

```diff
@@ -200,24 +184,119 @@ impl MultiValuedFastFieldWriter {
                             .iter()
                             .map(|val| *mapping.get(val).expect("Missing term ordinal"));
                         for val in remapped_vals {
-                            value_serializer.add_val(val)?;
+                            values.push(val);
                         }
                     }
                 }
             } else {
-                let val_min_max = minmax(self.vals.iter().cloned());
-                let (val_min, val_max) = val_min_max.unwrap_or((0u64, 0u64));
-                value_serializer =
-                    serializer.new_u64_fast_field_with_idx(self.field, val_min, val_max, 1)?;
                 for vals in self.get_ordered_values(doc_id_map) {
                     // sort values in case of remapped doc_ids?
                     for &val in vals {
-                        value_serializer.add_val(val)?;
+                        values.push(val);
                     }
                 }
             }
-            value_serializer.close_field()?;
+            let col = VecColumn::from(&values[..]);
+            serializer.create_auto_detect_u64_fast_field_with_idx(self.field, col, 1)?;
         }
         Ok(())
     }
 }
+
+pub(crate) struct MultivalueStartIndex<'a, C: Column> {
+    column: C,
+    doc_id_map: &'a DocIdMapping,
+    min_max_opt: Mutex<Option<(u64, u64)>>,
+}
+
+impl<'a, C: Column> MultivalueStartIndex<'a, C> {
+    pub fn new(column: C, doc_id_map: &'a DocIdMapping) -> Self {
+        assert_eq!(column.num_vals(), doc_id_map.num_old_doc_ids() as u64 + 1);
+        MultivalueStartIndex {
+            column,
+            doc_id_map,
+            min_max_opt: Mutex::default(),
+        }
+    }
+
+    fn minmax(&self) -> (u64, u64) {
+        if let Some((min, max)) = self.min_max_opt.lock().unwrap().clone() {
+            return (min, max);
+        }
+        let (min, max) = tantivy_bitpacker::minmax(self.iter()).unwrap_or((0u64, 0u64));
+        *self.min_max_opt.lock().unwrap() = Some((min, max));
+        (min, max)
+    }
+}
+impl<'a, C: Column> Column for MultivalueStartIndex<'a, C> {
+    fn get_val(&self, idx: u64) -> u64 {
```

Comment:
Estimations of the codecs sample in 5% steps and call `get_val`, so this is quite expensive (10x full-scan in total). But this is much less error prone than the approach I used elsewhere with the reverse computation. We could handle `get_vals` the same everywhere like here and precompute the sampled values once via `iter` and pass them to the codec estimates.

Reply:
I think before we were always picking the bitpacker... which is a bit sad considering the index of  multivalued stuff can benefit from the linear codec the most.

Reply:
Only in the initial creation, the merge code had autodetection